src/storage-client/src/storage_collections.rs

```diff
@@ -137,166 +111,10 @@ pub trait StorageCollections: Debug {
         id: Vec<GlobalId>,
     ) -> Result<Vec<CollectionFrontiers<Self::Timestamp>>, StorageError<Self::Timestamp>>;
 
-    /// Atomically gets and returns the frontiers of all active collections.
-    ///
-    /// A collection is "active" when it has a non empty frontier of read
-    /// capabilties.
-    fn active_collection_frontiers(&self) -> Vec<CollectionFrontiers<Self::Timestamp>>;
-
     /// Checks whether a collection exists under the given `GlobalId`. Returns
     /// an error if the collection does not exist.
     fn check_exists(&self, id: GlobalId) -> Result<(), StorageError<Self::Timestamp>>;
 
-    /// Returns aggregate statistics about the contents of the local input named
-    /// `id` at `as_of`.
-    async fn snapshot_stats(
-        &self,
-        id: GlobalId,
-        as_of: Antichain<Self::Timestamp>,
-    ) -> Result<SnapshotStats, StorageError<Self::Timestamp>>;
-
-    /// Returns aggregate statistics about the contents of the local input named
-    /// `id` at `as_of`.
-    ///
-    /// Note that this async function itself returns a future. We may
-    /// need to block on the stats being available, but don't want to hold a reference
-    /// to the controller for too long... so the outer future holds a reference to the
-    /// controller but returns quickly, and the inner future is slow but does not
-    /// reference the controller.
-    async fn snapshot_parts_stats(
-        &self,
-        id: GlobalId,
-        as_of: Antichain<Self::Timestamp>,
-    ) -> BoxFuture<'static, Result<SnapshotPartsStats, StorageError<Self::Timestamp>>>;
-
-    /// Update the given [`StorageTxn`] with the appropriate metadata given the
-    /// IDs to add and drop.
-    ///
-    /// The data modified in the `StorageTxn` must be made available in all
-    /// subsequent calls that require [`StorageMetadata`] as a parameter.
-    async fn prepare_state(
```

Comment:
`prepare_state` and `initialize_state` are also only meant to be on collections, the controller doesn't have persistent state anymore.